### DISCUSSION TITLE
Visual Studio Code C/C++ Extension gives the error "The language server has crashed 5 times in the last 3 minutes" and gives another error in the terminal "[Error - 4:25:28 pm] Connection to server got closed. Server will not be restarted."

### DISCUSSION THREAD

**Discussion by Chinmayahello** [microsoft/vscode-cpptools#7876] [Q&A]
Visual Studio Code C/C++ Extension gives the error "The language server has crashed 5 times in the last 3 minutes" and gives another error in the terminal "[Error - 4:25:28 pm] Connection to server got closed. Server will not be restarted."I have tried various versions of this extension ranging from 1.5.1 to 0.29 and this error is still not resolved.I am not tech savvy hence please use generic terms and give me step by step process according to  getting information about this error.I do not have any other informatio other than that I have a Windows 7 Servise Pack 1 OS.

**Comment by djairao** [comment]
From the error message, it appears that the backend server is having trouble starting up. From the error it appears that you are running on a server. Is that correct? That is, are you running the `cpptools` extension on a remote server/machine? If yes, what is the remote OS? Is it GNU/Linux?

**Reply by Chinmayahello** [reply]
I am not on a server.I am on my laptop and I get the error on that and for a fact I am using the GNU GCC Compiler form MinGW

**Reply by Colengms** [reply]
Hi @Chinmayahello .  Is there source code associated with repro'ing the crash?  Or, is there no source involved?  It's possible the issue is related to running a version of Windows 7 that is not fully up to date.  (I believe the latest service pack is 3).  I believe a fully patched Windows 7 should be supported.

**Reply by sean-mcmanus** [reply]
cpptools is crashing. Does this repro with any folder? Are you able to get it to not crash? It could be some issue with your Windows 7 OS and it looks like our extension may not be runnable due to missing required OS libraries. We don't test with Windows 7 and it's past it's end of life, so upgrading to Windows 10 is recommended if possible or possibly getting newer service pack updates.

**Reply by Chinmayahello** [reply]
I wanted to say that after some tries the extension finally works.I deleted my .vscode folder and completely uninstalled vs code .I reinstalled vs code but instead of downloading vs code I downloaded vs code insiders and it works! I do not the difference between vs code and vs code insiders so can you tell me the difference between them.For me it was completeley same
